File: dotnet/Aws.Commands/PublishMessage.cs
using System;
using System.Threading;
using System.Threading.Tasks;

using Microsoft.Extensions.Logging;

using Amazon.SimpleNotificationService;
using Amazon.SimpleNotificationService.Model;
using MediatR;

namespace Platform8.Aws.Commands
{
  public class PublishMessageRequest : INotification
  {
    public string Subject { get; set; }
    public string Topic { get; set; }
    public string Message { get; set; }
  }

  public class PublishMessageHandler : INotificationHandler<PublishMessageRequest>
  {
    private readonly IAmazonSimpleNotificationService snsService;
    private readonly ILogger<PublishMessageHandler> logger;
    public PublishMessageHandler(IAmazonSimpleNotificationService snsService, ILogger<PublishMessageHandler> logger)
    {
      this.snsService = snsService;
      this.logger = logger;
    }

    public async Task Handle(PublishMessageRequest request, CancellationToken cancellationToken)
    {
      var publishRequest = new PublishRequest
      {
        Subject = request.Subject,
        TargetArn = $"arn:aws:sns:us-west-1:101010101010:{request.Topic}", //TODO: Fix this...
        Message = request.Message
      };

      var result = await this.snsService.PublishAsync(publishRequest);

      this.logger.LogInformation($"Published TopicArn/MessageId: {publishRequest.TopicArn}/{result.MessageId}");      
    }
  }
}

File: packages/aws-commands/src/publishMessage.ts
import { Inject } from 'inversify-props';
import { PublishCommand, SNSClient,  } from "@aws-sdk/client-sns";
import { Command } from '@platform8/commands/src';

export interface PublishMessageRequest {
  subject?: string,
  message?: string,
  topic: string
}

export interface PublishMessageResponse { }

export class PublishMessageCommand implements Command<PublishMessageRequest, PublishMessageResponse> {

  @Inject("SNSClient")
  private snsClient!: SNSClient;

  async runAsync(params: PublishMessageRequest): Promise<any> {

    var sendParams = {
      Subject: params.subject,
      Message: params.message,
      TopicArn: `arn:aws:sns:us-west-1:101010101010:${params.topic}` //TODO: Fix this...
    };

    const command = new PublishCommand(sendParams);
    var result = await this.snsClient.send(command);

    console.log(`Published TopicArn/MessageId: ${sendParams.TopicArn}/${result.MessageId}`);
  }
}
